frontend/catalyst/pennylane_extensions.py

```diff
@@ -269,31 +318,125 @@ def circuit(x):
     >>> workflow(2.0)
     array(-3.14159265)
     """
-    methods = {"fd", "ps", "adj"}
-    if method is None:
-        method = "fd"
-    if method not in methods:
-        raise ValueError(
-            f"Invalid differentiation method '{method}'. "
-            f"Supported methods are: {' '.join(sorted(methods))}"
-        )
-    if method == "fd" and h is None:
-        h = 1e-7
-    if not (h is None or isinstance(h, numbers.Number)):
-        raise ValueError(f"Invalid h value ({h}). None or number was excpected.")
-    if argnum is None:
-        argnum = [0]
-    elif isinstance(argnum, int):
-        argnum = [argnum]
+    return Grad(_ensure_differentiable(f), grad_params=_check_grad_params(method, h, argnum))
+
+
+def jvp(f, params, tangents, *, method=None, h=None, argnum=None):
+    """A Jacobian-vector product for PennyLane/Catalyst.
+
+    This function allows the Jacobian-vector Product of a hybrid quantum-classical function to be
+    computed within the compiled program.
+
+    Args:
+        f(DifferentiableLike): Function-like object to calculate JVP for
+        params(List[Array]): List (or a tuble) of f's arguments specifying the point to calculate
+                             JVP at. A subset of these parameters are declared as
+                             differentiable by listing their indices in the ``argnum`` paramerer.
+        tangents(List[Array]): List (or a tuple) of tangent values to use in JVP. The list size and
+                               shapes must match the ones of differentiable params.
+        method(str): Differentiation method to use, same as in ``grad``.
```

Comment:
```suggestion
        method(str): Differentiation method to use, same as in :func:`~.grad`.
```

Best to always use cross referencing, so that a reader can easily access the other docstring if browsing on the web documentation